web/frontend/src/utils/analyticsUtil.ts

```diff
@@ -68,10 +91,12 @@ export const addChatInteraction = (
   llmResponse: string,
   recording_id: string,
   topk: TopKItem[] | undefined,
+  generationComplete: boolean,
 ): number => {
   let newChatIndex = -1;
   updateAnalyticsData((data) => {
     const newChat: ChatInteraction = {
+      llmResponseId: generateUUID(),
       timestamp: new Date().toISOString(),
       userPrompt,
       llmResponse,
```

Comment:
if we have the recording_id do we need to send the llm response and topK source item back? recording_id should be sufficient to have all these details right?

I am thinking how we can compress the payload for every click/hover event. The smaller the better.

Reply:
does the recording_id also contain the topK source items as well as llmResponse?